test/unit/org/apache/cassandra/db/guardrails/GuardrailsTest.java

```diff
@@ -129,7 +129,7 @@ public void testThresholdUsers() throws Throwable
 
         // value over fail threshold. An undefined user means that the check comes from a background process,
         // so we warn instead of failing to prevent interrupting that process.
-        assertWarns(() -> guard.guard(101, "z", null), "Failure: for z, 101 > 100");
+        assertFails(() -> guard.guard(101, "z", null), false, "Failure: for z, 101 > 100");
```

Comment:
Why is this Failing now instead of Warning?

Reply:
Now this is using the boolean `thrown` argument in `GuardrailsTester#assertFails` to indicate that this triggers the fail threshold but it shouldn't throw an exception (because of the null `ClientState`). That means that from client's perspective it only emits a client warning instead of failing, but it still emits an internal log message with `ERROR` level, and a `GuardrailEventType.ERROR` diagnostic event. So it's strictly an error but it doesn't abort the operation.

I have tried to improve the comment to better reflect this.

Reply:
Ok yes, now that makes sense. Thx